Assets/_Scripts/Enemy/Enemy.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.IO.LowLevel.Unsafe;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public abstract class Enemy : MonoBehaviour
+{
+    protected SpriteRenderer sr;
+    protected Rigidbody2D rb;
+    protected BoxCollider2D bc;
+    protected Animator anim;
+    
+    [SerializeField] private int EnemyHealth;
+    [SerializeField] private EnemyProjectile enemyProjectile;
+    public Transform enemyProjectileSpawn;
+    [SerializeField] private int projectileSpeed;
+    public float TimeToDestroy = 1;
+
+
+    protected virtual void Start()
+    {
+        sr = GetComponent<SpriteRenderer>();
+        rb = GetComponent<Rigidbody2D>();
+        bc = GetComponent<BoxCollider2D>();
+        anim = GetComponent<Animator>();
+
+        EnemyHealth = 2;
```

Comment:
Note that this will set whatever is configured in the unity editor for `EnemyHealth` to `2` whenever the game starts

Reply:
Yeah that's correct, we figured as the most times an enemy can be hit is twice (for the heavy enemy's) they didn't need more then 2 health. smaller enemies will take 2 damage while the heavies will only take 1, there for it takes two shot to kill them. We can always but this into a method where each enemy can set its own health but we didn't think it was necessary as for the reasons above. Let us know if you want us to approach this differently and we can change it.

Reply:
ohh I see - so when a player projectile hits light or medium enemies, those enemies need to take 2 damage, and when it hits a heavy enemy, it needs to take 1 damage. That works fine but my concern here is that this assignment is overriding what's being set in the unity editor. If for some reason we want to test and change `EnemyHealth` to 10 in unity, It'll just get set back to 2 here.

I agree with taking 2 damage but I'm thinking that this `EnemyHealth = 2` isn't needed here and can be set as a default where the field is defined and explicitly in the unity editor.

Reply:
I see what your saying, my bad I didn't realize this was set in the start, I thought we set it where it was declared at the top. We should change this.